src/status_im/extensions/capacities/components.cljs

```diff
@@ -0,0 +1,156 @@
+(ns status-im.extensions.capacities.components
+  (:require [status-im.ui.components.react :as react]
+            [status-im.chat.commands.impl.transactions :as transactions]
+            [status-im.ui.components.button.view :as button]
+            [re-frame.core :as re-frame]
+            [status-im.utils.platform :as platform]
+            [status-im.ui.components.list.views :as list]
+            [status-im.ui.components.checkbox.view :as checkbox]
+            [status-im.ui.components.colors :as colors]
+            [status-im.ui.components.icons.vector-icons :as icons]
+            [status-im.extensions.capacities.map :as map-component]))
+
+(defn button [{:keys [on-click enabled disabled] :as m} label]
+  [button/secondary-button (merge {:disabled? (or (when (contains? m :enabled) (or (nil? enabled) (false? enabled))) disabled)}
```

Comment:
Long line, hard for me to comprehend nested boolean expressions. Maybe multi-line would be easier to read here.